### PR TITLE
KT-21743 Method reference not correctly moved into parentheses

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-21743

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/MoveSuspiciousCallableReferenceIntoParenthesesInspection.kt [hunk 1]
```diff
@@ -43,26 +45,36 @@ class MoveSuspiciousCallableReferenceIntoParenthesesInspection : AbstractKotlinI
...
             val callableReference = callableReferenceExpression.callableReference
-            val resolvedCall = callableReference.resolveToCall(BodyResolveMode.FULL)
-            val receiverValue = resolvedCall?.let { it.extensionReceiver ?: it.dispatchReceiver }
-            return (receiverValue?.let { "${it.type}" } ?: "") + "::${callableReference.text}"
+            val receiverExpression = callableReferenceExpression.receiverExpression
+            val receiver = if (receiverExpression == null) {
+                ""
+            } else {
+                val descriptor = receiverExpression.getCallableDescriptor()
+                if (descriptor == null || descriptor is ValueParameterDescriptor)
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/inspections/MoveSuspiciousCallableReferenceIntoParenthesesInspection.kt]: This is suspicious place for me. Could you please explain why `ValueParameterDescriptor`, and only this descriptor, should be handled separately here?

Comment (t-kameyama) [reply idea/src/org/jetbrains/kotlin/idea/inspections/MoveSuspiciousCallableReferenceIntoParenthesesInspection.kt]: It is for lambda parameter. If remove ` || descriptor is ValueParameterDescriptor`, following tests fail.
- `defaultParameter.kt`
- `it.kt`
- `parameter.kt`
- `parameter2.kt`

Comment (mglukhikh) [reply idea/src/org/jetbrains/kotlin/idea/inspections/MoveSuspiciousCallableReferenceIntoParenthesesInspection.kt]: It should be not just "some" parameter, but this lambda parameter. See this fixing commit: https://github.com/JetBrains/kotlin/commit/b739db3c3573fb24daf1e97c42746d617dea4692.

**Comment by mglukhikh** [comment]
Thank you, I fixed & merged it.

**Pull request closed without merging**